htdocs/core/modules/modTakePos.class.php

```diff
@@ -203,6 +203,20 @@ public function __construct($db)
 		$this->rights[$r][3] = 0;
 		$this->rights[$r][4] = 'run';
 
+		$r++;
+		$this->rights[$r][0] = 50152;
+		$this->rights[$r][1] = 'Edit sales lines';
+		$this->rights[$r][2] = 'a';
+		$this->rights[$r][3] = 0;
+		$this->rights[$r][4] = 'editlines';
+
+		$r++;
+		$this->rights[$r][0] = 50153;
+		$this->rights[$r][1] = 'Edit ordered sales lines';
+		$this->rights[$r][2] = 'a';
+		$this->rights[$r][3] = 0;
+		$this->rights[$r][4] = 'editorderedlines';
```

Comment:
So it is a permission used when the POS is used to generate an order and not an invoice. In which case do we have this ?
With a common setup, the POS generate an invoice. Even when we enable the option "Enable features for Bar or Restaurant", document generated is "invoice". How do you generate "Order" ?

Reply:
TakePOS only generates invoices in Dolibarr even in restaurant bar mode.
The concept of order refers to printing the invoice (before being validated) in the kitchen to be able to prepare the food.
A printed line in the kitchen is marked as already printed in Dolibarr with $line->special_code and means 'already ordered'
In a restaurant it may be necessary to avoid erasing these already printed lines.

Reply:
So this is what i understood:
The second permission you add will be usefull if the option "Order Notes" is ON (if not, there is no order process so second permission will be useless). Right ?
After checking the code, it appears the "Order Notes" is available only if we choose a print method that is not browser. Why does this features "Order Notes" is not available when we use the browser ?

Reply:
No, "Order Notes" is another functionality. "Order Notes" is when you want a steak, but you want to add the "Very Done"  note. It is a functionality that extends "Order".
"Order notes" is avaiable only for kitchen printers, and that is because it is only available with a printing system that allows several printers, the browser allow only one printer.

On the other hand, several terms are currently unclear in TakePOS, especially with Spanish translations. Many thanks to the community, but there are terms that have to be revised according to the context. I have proposed changes in Transifex but they have not been published. Can I improve the terms and translations? How?

Reply:
Sorry. it is a condition on "Order notes", i was speaking on option "Order printers"

The update of special_code is done after a
`if ($action == "order"`...
line 691 of invoice.php

And the 
invoice.php?action=order
is done into TakeposPrintingOrder() js function

and this function by the "phone" interface or when the TAKEPOS_ORDER_PRINTERS is set. And to be able to set this option, we must set the printer TAKEPOS_PRINT_METHOD before, so option ""Order printers""

So it means that the permissions you want to add will be usefull only when the option ""Order printers"" is set. Right ?